pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -650,6 +650,12 @@ type CommonPrometheusFields struct {
 	// +optional
 	// +kubebuilder:validation:Minimum=60
 	MaximumStartupDurationSeconds *int32 `json:"maximumStartupDurationSeconds,omitempty"`
+
+	// EXPERIMENTAL List of scrape classes to expose to monitors and other scrape configs.
+	// This is experimental feature and might change in the future.
+	// +listType=map
+	// +listMapKey=name
+	ScrapeClasses []*ScrapeClass `json:"scrapeClasses,omitempty"`
```

Comment:
no need for a pointer?
```suggestion
	ScrapeClasses []ScrapeClass `json:"scrapeClasses,omitempty"`
```

Reply:
@simonpasquier this was introduced based on a suggestion provided by @ArthurSens: https://github.com/prometheus-operator/prometheus-operator/pull/6199#discussion_r1460690072

wdyt?

Reply:
The link talks about methods with pointer vs. value receivers which is a bit different than here.
If the API allows `[]*ScrapeClasses`, this would be a valid spec though useless:
```
spec:
  scrapeClasses:
  - null
  - name: foo
    tlsConfig: {...}
  - null
````

In this case, I'd rather favor the API consistency than optimization.

Reply:
No problems from my side, as mentioned in the review that was early optimization :)